src/pages/billing/List.js

```diff
@@ -241,11 +241,11 @@ export const BillingList = ({ queryParams: { selectedName } }) => {
   const loadAccounts = _.flow(
     withErrorReporting('Error loading billing accounts'),
     Utils.withBusyState(setIsLoadingAccounts)
-  )(async () => {
+  )(() => {
     if (Auth.hasBillingScope()) {
-      const accounts = await Ajax(signal).Billing.listAccounts()
-      const byAccountName = Object.fromEntries(_.map(acc => [acc.accountName, acc], accounts))
-      setBillingAccounts(byAccountName)
+      return Ajax(signal).Billing.listAccounts()
+        .then(_.keyBy('accountName'))
+        .then(setBillingAccounts)
```

Comment:
Functionally the same, but without the `.then` chaining, which we tend to avoid in favor of `async`/`await`:

```js
async () => {
  if (Auth.hasBillingScope()) {
    return _.flow(
      _.keyBy('accountName'),
      setBillingAccounts
    )(await Ajax(signal).Billing.listAccounts())
  }
}
```

I can't decide if I think this looks ickier than what you have here, though.

Reply:
I prefer mine - it reads more linearly.